SourceX/DiabloUI/diabloui.cpp

```diff
@@ -454,19 +454,19 @@ void LoadHeros()
 	LoadArt("ui_art\\heros.pcx", &ArtHero);
 
 	const int portraitHeight = 76;
-	int portraitOrder[NUM_CLASSES + 1] = { 0, 1, 2, 2, 1, 0, 3 };
+	int portraitOrder[static_cast<std::size_t>(HeroClass::ClassCount) + 1] = { 0, 1, 2, 2, 1, 0, 3 };

```

Comment:
How come this now requires us to cast the enum when using them as indexes?

Reply:
`enum class` is not implicitly convertible to integral types

Reply:
sad, can we make it so :)

Reply:
That's a feature, it's to prevent accidental conversions to/from int.

Reply:
It's one of the unfortunate things about indexing with them into a raw array. The `std::map` avoids this but... well... we already discussed that option 😛

Reply:
> That's a feature, it's to prevent accidental conversions to/from int.

Wait but that is exactly what we use them for :D
Originally they were all written as `#define ATTRIB_STR 0` :D

Reply:
I think the core here is "accidental". When you add the `static_cast` you are being explicit about the intention of using it as a number.

Reply:
```cpp
enum : permissive <with::count> {
}
```

Reply:
What about `std::unordere...`... ah, forget it 😆

Reply:
Where is the discussion about using a map?

Reply:
Discord